### ISSUE TITLE
Issue with compiling 

### ISSUE THREAD

**Issue by disputed1** [Openwsman/openwsman#102]
-- The C compiler identification is GNU 4.9.2
-- The CXX compiler identification is GNU 4.9.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Version 2.6.3
-- Package Version "2.6"
-- Plugin API Version "2.2"
-- Building for x86_64
-- Libraries will be installed in /usr/local/lib64
-- Configuration will be installed in /usr/local/etc/openwsman
-- building PAM authentication
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.0.1t") 
-- Building Ruby bindings
-- ************  WARNING  *************
-- Didn't Find Ruby Headers... Not Building Ruby Bindings
-- ************************************
-- Building Python bindings
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython2.7.so (found suitable version "2.7.9", minimum required is "2.6") 
-- Found PythonInterp: /usr/bin/python2 (found suitable version "2.7.9", minimum required is "2.6") 
-- Found PythonLibs...
-- Building Python...
-- Found Perl: /usr/bin/perl (found version "5.20.2") 
-- Building Perl bindings
-- Could NOT find Java (missing:  Java_JAR_EXECUTABLE Java_JAVAC_EXECUTABLE Java_JAVAH_EXECUTABLE Java_JAVADOC_EXECUTABLE) (found version "1.7.0.121")
CMake Warning at CMakeLists.txt:222 (MESSAGE):
   Java compiler not found

-- Found SWIG: /usr/bin/swig2.0 (found version "2.0.12") 
-- SWIG 2.0.12 found at /usr/bin/swig2.0
-- Python executable: /usr/bin/python2
-- Python lib dir: /usr/lib/python2.7/dist-packages
-- Perl executable: /usr/bin/perl
-- Perl core dir: /usr/lib/x86_64-linux-gnu/perl/5.20/CORE
-- Perl vendor arch dir: /usr/lib/x86_64-linux-gnu/perl5/5.20
-- Perl vendor lib dir: /usr/share/perl5
-- Found CURL: /usr/lib/x86_64-linux-gnu/libcurl.so (found version "7.38.0") 
-- Found LibXml2: /usr/lib/x86_64-linux-gnu/libxml2.so (found version "2.9.1") 
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Looking for crypt.h
-- Looking for crypt.h - found
-- Looking for ctype.h
-- Looking for ctype.h - found
-- Looking for CUnit/Basic.h
-- Looking for CUnit/Basic.h - not found
-- Looking for dirent.h
-- Looking for dirent.h - found
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for ifaddrs.h
-- Looking for ifaddrs.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for net/if_dl.h
-- Looking for net/if_dl.h - not found
-- Looking for net/if.h
-- Looking for net/if.h - found
-- Looking for pam/pam_appl.h
-- Looking for pam/pam_appl.h - not found
-- Looking for pam/pam_misc.h
-- Looking for pam/pam_misc.h - not found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for security/pam_appl.h
-- Looking for security/pam_appl.h - found
-- Looking for security/pam_misc.h
-- Looking for security/pam_misc.h - found
-- Looking for stdarg.h
-- Looking for stdarg.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for sys/ioctl.h
-- Looking for sys/ioctl.h - found
-- Looking for sys/resource.h
-- Looking for sys/resource.h - found
-- Looking for sys/select.h
-- Looking for sys/select.h - found
-- Looking for sys/sendfile.h
-- Looking for sys/sendfile.h - found
-- Looking for sys/signal.h
-- Looking for sys/signal.h - found
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for sys/sockio.h
-- Looking for sys/sockio.h - not found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for vararg.h
-- Looking for vararg.h - not found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for getifaddrs
-- Looking for getifaddrs - found
-- Looking for alloca
-- Looking for alloca - not found
-- Looking for alloca.h
-- Looking for alloca.h - found
-- Looking for bcopy
-- Looking for bcopy - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of in_addr_t
-- Check size of in_addr_t - failed
-- Check type in_addr_t as HAVE_IN_ADDR_T
-- Check type in_port_t as HAVE_IN_PORT_T
-- Check type sa_family_t as HAVE_SA_FAMILY_T
-- Check type int as HAVE_INT
-- Check type long as HAVE_LONG
-- Check type long long as HAVE_LONG_LONG
-- Check type short as HAVE_SHORT
-- Performing Test HAVE_SA_LEN
-- Performing Test HAVE_SA_LEN - Failed
-- Performing Test HAVE_IPV6
-- Performing Test HAVE_IPV6 - Success
-- SSL_LIB is at ><
-- Package file openwsman-2.6.3
--    ************ NOTE: ************
--    Warnings occurred during cmake configuration... Please see output
--    *******************************
-- Configuring done
-- Generating done
-- Build files have been written to: /home/openwsman/build
root@debian:/home/openwsman/build# make
Scanning dependencies of target wsman_curl_client_transport
[  1%] Building C object src/lib/CMakeFiles/wsman_curl_client_transport.dir/wsman-client-transport.o
[  2%] Building C object src/lib/CMakeFiles/wsman_curl_client_transport.dir/wsman-curl-client-transport.o
Linking C shared library libwsman_curl_client_transport.so
[  2%] Built target wsman_curl_client_transport
Scanning dependencies of target wsman_client
[  3%] Building C object src/lib/CMakeFiles/wsman_client.dir/wsman-client.o
Linking C shared library libwsman_client.so
[  3%] Built target wsman_client
Scanning dependencies of target wsman
[  4%] Building C object src/lib/CMakeFiles/wsman.dir/u/buf.o
[  5%] Building C object src/lib/CMakeFiles/wsman.dir/u/log.o
[  6%] Building C object src/lib/CMakeFiles/wsman.dir/u/memory.o
[  7%] Building C object src/lib/CMakeFiles/wsman.dir/u/misc.o
[  8%] Building C object src/lib/CMakeFiles/wsman.dir/u/uri.o
[  9%] Building C object src/lib/CMakeFiles/wsman.dir/u/uuid.o
[ 10%] Building C object src/lib/CMakeFiles/wsman.dir/u/lock.o
[ 11%] Building C object src/lib/CMakeFiles/wsman.dir/u/md5.o
[ 12%] Building C object src/lib/CMakeFiles/wsman.dir/u/strings.o
[ 13%] Building C object src/lib/CMakeFiles/wsman.dir/u/list.o
[ 14%] Building C object src/lib/CMakeFiles/wsman.dir/u/hash.o
[ 15%] Building C object src/lib/CMakeFiles/wsman.dir/u/base64.o
[ 16%] Building C object src/lib/CMakeFiles/wsman.dir/u/iniparser.o
[ 17%] Building C object src/lib/CMakeFiles/wsman.dir/u/debug.o
[ 18%] Building C object src/lib/CMakeFiles/wsman.dir/u/uerr.o
[ 19%] Building C object src/lib/CMakeFiles/wsman.dir/u/uoption.o
[ 20%] Building C object src/lib/CMakeFiles/wsman.dir/u/gettimeofday.o
[ 21%] Building C object src/lib/CMakeFiles/wsman.dir/u/syslog.o
[ 22%] Building C object src/lib/CMakeFiles/wsman.dir/u/pthreadx_win32.o
[ 23%] Building C object src/lib/CMakeFiles/wsman.dir/u/os.o
[ 24%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-libxml2-binding.o
[ 25%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-xml.o
[ 26%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-epr.o
[ 27%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-key-value.o
[ 28%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-filter.o
[ 29%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-dispatcher.o
[ 30%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-soap.o
[ 31%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-faults.o
[ 32%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-xml-serialize.o
[ 33%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-soap-envelope.o
[ 34%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-debug.o
[ 35%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-soap-message.o
[ 36%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-subscription-repository.o
[ 37%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-event-pool.o
[ 38%] Building C object src/lib/CMakeFiles/wsman.dir/wsman-cimindication-processor.o
Linking C shared library libwsman.so
[ 38%] Built target wsman
[ 39%] Generating openwsman_wrap.c
Creating wrapper code for Python ...Done.
Scanning dependencies of target pywsman
[ 40%] Building C object bindings/python/CMakeFiles/pywsman.dir/openwsman_wrap.o
[ 41%] Building C object bindings/python/CMakeFiles/pywsman.dir/__/openwsman.o
Linking C shared library _pywsman.so
[ 41%] Built target pywsman
[ 42%] Generating openwsman_wrap.c
Creating wrapper code for Perl .../home/openwsman/bindings/wsman-xml.i:564: Warning 314: 'next' is a perl keyword
Done.
Scanning dependencies of target plwsman
[ 43%] Building C object bindings/perl/CMakeFiles/plwsman.dir/openwsman_wrap.o
[ 44%] Building C object bindings/perl/CMakeFiles/plwsman.dir/__/openwsman.o
Linking C shared library openwsman.so
[ 44%] Built target plwsman
Scanning dependencies of target wsman_server
[ 45%] Building C object src/lib/CMakeFiles/wsman_server.dir/wsman-server.o
[ 46%] Building C object src/lib/CMakeFiles/wsman_server.dir/wsman-plugins.o
[ 47%] Building C object src/lib/CMakeFiles/wsman_server.dir/wsman-server-api.o
Linking C shared library libwsman_server.so
[ 47%] Built target wsman_server
Scanning dependencies of target test_list
[ 48%] Building C object src/lib/test/CMakeFiles/test_list.dir/test_list.o
Linking C executable test_list
[ 48%] Built target test_list
Scanning dependencies of target test_md5
[ 49%] Building C object src/lib/test/CMakeFiles/test_md5.dir/test_md5.o
Linking C executable test_md5
[ 49%] Built target test_md5
Scanning dependencies of target test_string
[ 50%] Building C object src/lib/test/CMakeFiles/test_string.dir/test_string.o
Linking C executable test_string
[ 50%] Built target test_string
Scanning dependencies of target wsman_clientpp
[ 51%] Building CXX object src/cpp/CMakeFiles/wsman_clientpp.dir/OpenWsmanClient.o
[ 52%] Building CXX object src/cpp/CMakeFiles/wsman_clientpp.dir/WsmanEPR.o
[ 53%] Building CXX object src/cpp/CMakeFiles/wsman_clientpp.dir/WsmanFilter.o
[ 54%] Building CXX object src/cpp/CMakeFiles/wsman_clientpp.dir/WsmanOptions.o
Linking CXX shared library libwsman_clientpp.so
[ 54%] Built target wsman_clientpp
Scanning dependencies of target wsman_identify_plugin
[ 55%] Building C object src/plugins/identify/CMakeFiles/wsman_identify_plugin.dir/identify.o
[ 56%] Building C object src/plugins/identify/CMakeFiles/wsman_identify_plugin.dir/identify_stubs.o
Linking C shared library libwsman_identify_plugin.so
[ 56%] Built target wsman_identify_plugin
Scanning dependencies of target wsman_test
[ 57%] Building C object src/plugins/wsman/test/CMakeFiles/wsman_test.dir/wsman_test.o
[ 58%] Building C object src/plugins/wsman/test/CMakeFiles/wsman_test.dir/wsman_test_stubs.o
Linking C shared library libwsman_test.so
[ 58%] Built target wsman_test
Scanning dependencies of target redirect
[ 59%] Building C object src/plugins/redirect/CMakeFiles/redirect.dir/redirect.o
[ 60%] Building C object src/plugins/redirect/CMakeFiles/redirect.dir/redirect_stubs.o
Linking C shared library libredirect.so
[ 60%] Built target redirect
Scanning dependencies of target wsman_cim_plugin
[ 61%] Building C object src/plugins/cim/CMakeFiles/wsman_cim_plugin.dir/sfcc-interface.o
[ 62%] Building C object src/plugins/cim/CMakeFiles/wsman_cim_plugin.dir/cim_data.o
[ 63%] Building C object src/plugins/cim/CMakeFiles/wsman_cim_plugin.dir/cim_data_stubs.o
Linking C shared library libwsman_cim_plugin.so
[ 63%] Built target wsman_cim_plugin
Scanning dependencies of target openwsmand
[ 63%] Building C object src/server/CMakeFiles/openwsmand.dir/shttpd/string.o
In file included from /home/openwsman/src/server/shttpd/std_includes.h:37:0,
                 from /home/openwsman/src/server/shttpd/defs.h:16,
                 from /home/openwsman/src/server/shttpd/string.c:11:
**/home/openwsman/src/server/shttpd/compat_unix.h:30:0: warning: "SSL_LIB" redefined
 #define SSL_LIB    "libssl.so"
 ^
<command-line>:0:0: note: this is the location of the previous definition
In file included from /home/openwsman/src/server/shttpd/string.c:11:0:
/home/openwsman/src/server/shttpd/defs.h:21:20: fatal error: config.h: No such file or directory
 #include "config.h"
                    ^
compilation terminated.
src/server/CMakeFiles/openwsmand.dir/build.make:54: recipe for target 'src/server/CMakeFiles/openwsmand.dir/shttpd/string.o' failed
make[2]: *** [src/server/CMakeFiles/openwsmand.dir/shttpd/string.o] Error 1
CMakeFiles/Makefile2:1844: recipe for target 'src/server/CMakeFiles/openwsmand.dir/all' failed
make[1]: *** [src/server/CMakeFiles/openwsmand.dir/all] Error 2
Makefile:147: recipe for target 'all' failed
make: *** [all] Error 2**

On Debian Jessie

**Comment by beave** [comment]
Hello... 

It appears to be fixed with this pull:

https://github.com/Openwsman/openwsman/pull/100/commits/b9fd3352af2b50893a7fa42ec3c1ccc184e74bd5

Please consider merging :)

**Comment by babacoders** [comment]
I am also facing the similar issue:

```
root@debian:/tmp/openwsman-2.6.3/build# make install
/usr/bin/cmake -H/tmp/openwsman-2.6.3 -B/tmp/openwsman-2.6.3/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /tmp/openwsman-2.6.3/build/CMakeFiles /tmp/openwsman-2.6.3/build/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/tmp/openwsman-2.6.3/build'
make -f src/lib/CMakeFiles/wsman_curl_client_transport.dir/build.make src/lib/CMakeFiles/wsman_curl_client_transport.dir/depend
make[2]: Entering directory '/tmp/openwsman-2.6.3/build'
cd /tmp/openwsman-2.6.3/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/openwsman-2.6.3 /tmp/openwsman-2.6.3/src/lib /tmp/openwsman-2.6.3/build /tmp/openwsman-2.6.3/build/src/lib /tmp/openwsman-2.6.3/build/src/lib/CMakeFiles/wsman_curl_client_transport.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/tmp/openwsman-2.6.3/build'
make -f src/lib/CMakeFiles/wsman_curl_client_transport.dir/build.make src/lib/CMakeFiles/wsman_curl_client_transport.dir/build
make[2]: Entering directory '/tmp/openwsman-2.6.3/build'
[  0%] Building C object src/lib/CMakeFiles/wsman_curl_client_transport.dir/wsman-curl-client-transport.o
cd /tmp/openwsman-2.6.3/build/src/lib && /usr/bin/cc  -DHAVE_CONFIG_H -Dwsman_curl_client_transport_EXPORTS -I/usr/include/x86_64-linux-gnu -I/usr/include/libxml2 -I/tmp/openwsman-2.6.3/include -I/tmp/openwsman-2.6.3 -I/tmp/openwsman-2.6.3/build  -g -DIGNORE_DUPLICATE_ID -fPIC   -DPACKAGE_PLUGIN_DIR="\"/usr//lib/x86_64-linux-gnu/openwsman/plugins\"" -o CMakeFiles/wsman_curl_client_transport.dir/wsman-curl-client-transport.o   -c /tmp/openwsman-2.6.3/src/lib/wsman-curl-client-transport.c
/tmp/openwsman-2.6.3/src/lib/wsman-curl-client-transport.c: In function ‘ssl_certificate_thumbprint_verify_callback’:
/tmp/openwsman-2.6.3/src/lib/wsman-curl-client-transport.c:244:18: error: dereferencing pointer to incomplete type ‘X509_STORE_CTX {aka struct x509_store_ctx_st}’
  X509 *cert = ctx->cert;
                  ^~
src/lib/CMakeFiles/wsman_curl_client_transport.dir/build.make:89: recipe for target 'src/lib/CMakeFiles/wsman_curl_client_transport.dir/wsman-curl-client-transport.o' failed
make[2]: *** [src/lib/CMakeFiles/wsman_curl_client_transport.dir/wsman-curl-client-transport.o] Error 1
make[2]: Leaving directory '/tmp/openwsman-2.6.3/build'
CMakeFiles/Makefile2:1551: recipe for target 'src/lib/CMakeFiles/wsman_curl_client_transport.dir/all' failed
make[1]: *** [src/lib/CMakeFiles/wsman_curl_client_transport.dir/all] Error 2
make[1]: Leaving directory '/tmp/openwsman-2.6.3/build'
Makefile:163: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by babacoders** [comment]
This is what I did from a fresh 

```
root@debian:~# git clone https://github.com/Openwsman/openwsman
Cloning into 'openwsman'...
remote: Counting objects: 17857, done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 17857 (delta 1), reused 5 (delta 1), pack-reused 17847
Receiving objects: 100% (17857/17857), 9.11 MiB | 841.00 KiB/s, done.
Resolving deltas: 100% (11755/11755), done.
root@debian:~# cd openwsman/
root@debian:~/openwsman# mkdir build
root@debian:~/openwsman# cd build/
root@debian:~/openwsman/build# cmake ..
-- The C compiler identification is GNU 6.3.0
-- The CXX compiler identification is GNU 6.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Version 2.6.4
-- Package Version "2.6"
-- Plugin API Version "2.2"
-- Building for x86_64
-- Libraries will be installed in /usr/local/lib64
-- Configuration will be installed in /usr/local/etc/openwsman
-- building PAM authentication
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.1.0f")
-- Building Ruby bindings
-- ************  WARNING  *************
-- Didn't Find Ruby Headers... Not Building Ruby Bindings
-- ************************************
-- Building Python bindings
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython2.7.so (found suitable version "2.7.13", minimum required is "2.6")
-- Found PythonInterp: /usr/bin/python2 (found suitable version "2.7.13", minimum required is "2.6")
-- Found PythonLibs...
-- Building Python...
-- Found Perl: /usr/bin/perl (found version "5.24.1")
-- Building Perl bindings
-- Found Java: /usr/bin/java (found version "1.8.0.151")
-- Found JNI: /usr/lib/jvm/default-java/jre/lib/amd64/libjawt.so
-- Building Java bindings
-- Found SWIG: /usr/bin/swig3.0 (found version "3.0.10")
-- SWIG 3.0.10 found at /usr/bin/swig3.0
-- Python executable: /usr/bin/python2
-- Python lib dir: /usr/lib/python2.7/dist-packages
-- Java compiler: /usr/bin/javac
-- Java runtime: /usr/bin/java
-- Java archiver: /usr/bin/jar
-- Java archive name: openwsman-x86_64-2.6.4.jar
-- Perl executable: /usr/bin/perl
-- Perl core dir: /usr/lib/x86_64-linux-gnu/perl/5.24/CORE
-- Perl vendor arch dir: /usr/lib/x86_64-linux-gnu/perl5/5.24
-- Perl vendor lib dir: /usr/share/perl5
-- Found CURL: /usr/lib/x86_64-linux-gnu/libcurl.so (found version "7.52.1")
-- Found LibXml2: /usr/lib/x86_64-linux-gnu/libxml2.so (found version "2.9.4")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Looking for crypt.h
-- Looking for crypt.h - found
-- Looking for ctype.h
-- Looking for ctype.h - found
-- Looking for CUnit/Basic.h
-- Looking for CUnit/Basic.h - not found
-- Looking for dirent.h
-- Looking for dirent.h - found
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for ifaddrs.h
-- Looking for ifaddrs.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for net/if_dl.h
-- Looking for net/if_dl.h - not found
-- Looking for net/if.h
-- Looking for net/if.h - found
-- Looking for pam/pam_appl.h
-- Looking for pam/pam_appl.h - not found
-- Looking for pam/pam_misc.h
-- Looking for pam/pam_misc.h - not found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for security/pam_appl.h
-- Looking for security/pam_appl.h - found
-- Looking for security/pam_misc.h
-- Looking for security/pam_misc.h - found
-- Looking for stdarg.h
-- Looking for stdarg.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for sys/ioctl.h
-- Looking for sys/ioctl.h - found
-- Looking for sys/resource.h
-- Looking for sys/resource.h - found
-- Looking for sys/select.h
-- Looking for sys/select.h - found
-- Looking for sys/sendfile.h
-- Looking for sys/sendfile.h - found
-- Looking for sys/signal.h
-- Looking for sys/signal.h - found
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for sys/sockio.h
-- Looking for sys/sockio.h - not found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for vararg.h
-- Looking for vararg.h - not found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for getifaddrs
-- Looking for getifaddrs - found
-- Looking for alloca
-- Looking for alloca - not found
-- Looking for alloca.h
-- Looking for alloca.h - found
-- Looking for bcopy
-- Looking for bcopy - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of in_addr_t
-- Check size of in_addr_t - failed
-- Check type in_addr_t as HAVE_IN_ADDR_T
-- Check type in_port_t as HAVE_IN_PORT_T
-- Check type sa_family_t as HAVE_SA_FAMILY_T
-- Check type int as HAVE_INT
-- Check type long as HAVE_LONG
-- Check type long long as HAVE_LONG_LONG
-- Check type short as HAVE_SHORT
-- Performing Test HAVE_SA_LEN
-- Performing Test HAVE_SA_LEN - Failed
-- Performing Test HAVE_IPV6
-- Performing Test HAVE_IPV6 - Success
-- SSL_LIB is at ><
-- Package file openwsman-2.6.4
--    ************ NOTE: ************
--    Warnings occurred during cmake configuration... Please see output
--    *******************************
-- Configuring done
-- Generating done
-- Build files have been written to: /root/openwsman/build
root@debian:~/openwsman/build# make
Scanning dependencies of target wsman_curl_client_transport
[  1%] Building C object src/lib/CMakeFiles/wsman_curl_client_transport.dir/wsman-client-transport.o
[  1%] Building C object src/lib/CMakeFiles/wsman_curl_client_transport.dir/wsman-curl-client-transport.o
/root/openwsman/src/lib/wsman-curl-client-transport.c: In function ‘ssl_certificate_thumbprint_verify_callback’:
/root/openwsman/src/lib/wsman-curl-client-transport.c:244:18: error: dereferencing pointer to incomplete type ‘X509_STORE_CTX {aka struct x509_store_ctx_st}’
  X509 *cert = ctx->cert;
                  ^~
src/lib/CMakeFiles/wsman_curl_client_transport.dir/build.make:86: recipe for target 'src/lib/CMakeFiles/wsman_curl_client_transport.dir/wsman-curl-client-transport.o' failed
make[2]: *** [src/lib/CMakeFiles/wsman_curl_client_transport.dir/wsman-curl-client-transport.o] Error 1
CMakeFiles/Makefile2:1548: recipe for target 'src/lib/CMakeFiles/wsman_curl_client_transport.dir/all' failed
make[1]: *** [src/lib/CMakeFiles/wsman_curl_client_transport.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by kkaempf** [comment]
Fixed with [2.6.6](https://github.com/Openwsman/openwsman/releases/tag/v2.6.6) or later.
